doc/ps_design_interactions_among_master_worker_ps.md

```diff
@@ -0,0 +1,54 @@
+# ElasticDL Parameter Server Design Doc
+## Parameter Server
+
+
+## Interactions among Master, PS and Worker
+
+The following events involve interactions among the master, workers and PS:
+
+* The master starts PS.
+* Initialization of parameters in PS.
+* Relaunch of PS.
+* Workers get model parameters from PS.
+* Workers push gradients to PS.
+* PS reports submodel version to the master.
+* The master tells PS to save checkpoint.
+
+
+### The master starts PS
+When an ElasticDL task starts, `master.main` is responsible for starting PS. After starting PS, `master.main` starts the master and workers, and tells them the endpoints of PS.
```

Comment:
For multiple PSs with embedding replicas, the master needs to tell every PS the corresponding replica PS endpoints.
The domain name for PS should be fixed using Kubernetes services. In this way, if a PS pod is relaunched, it will use the same domain name. Workers and other PSs can still use the same endpoints for GRPC calls.